### ISSUE TITLE
Bug: API No Connection 

### ISSUE THREAD

**Issue by dampney** [OpenWonderLabs/homebridge-switchbot#1139]
### Describe the Bug

Errors with API connection. No devices found.

### To Reproduce

Uninstalled Plugin due to other bugs, removed bridge from HomeKit.
Reset token in SB app
Reinstalled plugin, added new token / secret, child bridge added to HomeKit.
Restarted HB.

### Expected Behavior

Have reinstalled the SB plugin half a dozen times over the last few weeks to months due to bugs with this plugin. Have basically given up with dealing with the issues. Now wanting the plugin to work for some simple IR lights instead of relying on major devices (AC, multiple bots etc).

### Relevant Log Output

```shell
[12/17/2024, 10:40:27 PM] [SwitchBot] Failed to get d evices: getaddrinfo ENOTFOUND api.switch-bot.com
[12/17/2024, 10:40:27 PM] [SwitchBot] Failed to get d evices: getaddrinfo ENOTFOUND api.switch-bot.com
[12/17/2024, 10:40:27 PM] [SwitchBot] Failed to get d evices: getaddrinfo ENOTFOUND api.switch-bot.com
112/17/2024, 10:40:27 PM] [SwitchBot] Failed to get d evices: getaddrinfo ENOTFOUND api.switch-bot.com
112/17/2024, 10:40:27 PM] [SwitchBot] Failed to get d evices: getaddrinfo ENOTFOUND api.switch-bot.com
```

### Config for homebridge-switchbot

```shell
{    "name": "SwitchBot",    "credentials": {        "token": "xxx",        "secret": "xxx",        "notice": "Keep your Token & Secret a secret!"    },    "_bridge": {        "username": "0E:D7:1B:2F:23:63",        "port": 47836    },    "platform": "SwitchBot"}
```

### Screenshots

![IMG_8426](https://github.com/user-attachments/assets/e8c0ea3f-96b8-4887-9769-5cf29c845db8)
![IMG_0015](https://github.com/user-attachments/assets/3c43a470-19db-4429-824b-09545a6f49ab)

### Device and Model

Mac Mini Late 

### Node.js Version

V20.10.p

### NPM Version

TBC

### Homebridge Version

V1.8.5

### Homebridge Switchbot Plugin Version

V4.1.5

### Homebridge Config UI X Plugin Version

V4.65.1

### Operating System

macOS

**Comment by blanky** [comment]
Having the same issues, using Cloudflare as DNS.
Doing a dig with Cloudflare and Google:

```
(base) ➜  ~ dig api.switch-bot.com

; <<>> DiG 9.10.6 <<>> api.switch-bot.com
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 62340
;; flags: qr rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 1, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 1232
;; QUESTION SECTION:
;api.switch-bot.com.		IN	A

;; ANSWER SECTION:
api.switch-bot.com.	3180	IN	CNAME	api.switchbot.net.

;; AUTHORITY SECTION:
switchbot.net.		480	IN	SOA	ns-1478.awsdns-56.org. awsdns-hostmaster.amazon.com. 1 7200 900 1209600 86400

;; Query time: 14 msec
;; SERVER: 1.1.1.1#53(1.1.1.1)
;; WHEN: Tue Dec 17 22:29:58 AEST 2024
;; MSG SIZE  rcvd: 160

(base) ➜  ~ dig @8.8.8.8 api.switch-bot.com

; <<>> DiG 9.10.6 <<>> @8.8.8.8 api.switch-bot.com
; (1 server found)
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 37095
;; flags: qr rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 1, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 512
;; QUESTION SECTION:
;api.switch-bot.com.		IN	A

;; ANSWER SECTION:
api.switch-bot.com.	426	IN	CNAME	api.switchbot.net.

;; AUTHORITY SECTION:
switchbot.net.		592	IN	SOA	ns-1478.awsdns-56.org. awsdns-hostmaster.amazon.com. 1 7200 900 1209600 86400

;; Query time: 27 msec
;; SERVER: 8.8.8.8#53(8.8.8.8)
;; WHEN: Tue Dec 17 22:30:09 AEST 2024
;; MSG SIZE  rcvd: 160
```
Interestingly, if I hop on a VPN to the USA it seems to work (am based in Australia).
```
(base) ➜  ~ dig api.switch-bot.com

; <<>> DiG 9.10.6 <<>> api.switch-bot.com
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 64557
;; flags: qr rd ra; QUERY: 1, ANSWER: 5, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 1232
;; QUESTION SECTION:
;api.switch-bot.com.		IN	A

;; ANSWER SECTION:
api.switch-bot.com.	3600	IN	CNAME	api.switchbot.net.
api.switchbot.net.	300	IN	CNAME	d-99k5evl0bf.execute-api.us-east-1.amazonaws.com.
d-99k5evl0bf.execute-api.us-east-1.amazonaws.com. 60 IN	A 34.203.59.42
d-99k5evl0bf.execute-api.us-east-1.amazonaws.com. 60 IN	A 54.208.27.114
d-99k5evl0bf.execute-api.us-east-1.amazonaws.com. 60 IN	A 34.227.3.41

;; Query time: 212 msec
;; SERVER: 10.0.0.243#53(10.0.0.243)
;; WHEN: Tue Dec 17 22:32:42 AEST 2024
;; MSG SIZE  rcvd: 185
```

That's about as far as I got, DNS is not my forte.

**Comment by domre77** [comment]
Same here. Nothing but issues with the API for months. Had to switch to BLE, which is unreliable but works most of the time.

**Comment by domre77** [comment]
Marked as stale? I know it’s automatic, but this a major bug that completely breaks the API part of the plugin, and it’s persisted for months. I understand people are busy, and I thank them for there work up until now, but, I read this as the big is not being fixed?

Maybe it’s time to move to Home Assistant for me I think.

**Comment by dampney** [comment]
@domre77, I just starting testing the Switchbot integration on HA.


TL;DR
SB into HA into HomeKit. Works well so far, not all devices are supported, AC is thermostat only, not Heater-Cooler.

- - -

I'm seeing IR lights, IR AC and Hub 2 temp/humidity sensors but not any bots, not a IR on/off switch and no Hub 2 light sensor / buttons.

My AC is a split-system with the Hub 2 IR blaster. The HA integration into HomeKit has worked reliable so far. 

There seems to be no support for Heater-Cooler, so HA pushes the device into HK as a thermostat.

A thermostat is fine but there is no "Off" selection within the controls but a separate on/off switch.

It's easy to change this switch into a fan in HomeKit so the device shows up under Climate controls.

The SB integration in HA does connect via the API - so a lack of local support if the internet goes down (thanks NBN).

I had relied on the SB Homebridge plugin for several years now due to Matters lack of controls and SB bugs.

Hope this helps

**Comment by rmclean45** [comment]
Skipping the setup config GUI and just creating the JSON Config as;

`{
  "platforms": [
    {
      "platform": "SwitchBot",
      "name": "SwitchBot",
      "credentials": {
        "token": "xxxxxxxxxx",
        "secret": "yyyyyyyyyyyy"
      },
      "options": {
        "logging": "standard"
      }
    }
  ]
}`

This completely solved the issue for me, the GUI config outputs something unuseable that fails validation upon install.

**Comment by domre77** [comment]
> [@domre77](https://github.com/domre77), I just starting testing the Switchbot integration on HA.
> 
> TL;DR SB into HA into HomeKit. Works well so far, not all devices are supported, AC is thermostat only, not Heater-Cooler.
> 
> I'm seeing IR lights, IR AC and Hub 2 temp/humidity sensors but not any bots, not a IR on/off switch and no Hub 2 light sensor / buttons.
> 
> My AC is a split-system with the Hub 2 IR blaster. The HA integration into HomeKit has worked reliable so far.
> 
> There seems to be no support for Heater-Cooler, so HA pushes the device into HK as a thermostat.
> 
> A thermostat is fine but there is no "Off" selection within the controls but a separate on/off switch.
> 
> It's easy to change this switch into a fan in HomeKit so the device shows up under Climate controls.
> 
> The SB integration in HA does connect via the API - so a lack of local support if the internet goes down (thanks NBN).
> 
> I had relied on the SB Homebridge plugin for several years now due to Matters lack of controls and SB bugs.
> 
> Hope this helps

I’ve switched to Home Assistant and haven’t looked back since. Should have done it years ago. The Bluetooth Switchbot integration hasn’t failed me once, and I’m encouraged by Switchbot’s announcement that they are extending official support for Home Assistant. Even as just a HomeKit bridge I prefer HA over Homebridge, but in general I think the Homebridge project is dying a slow death of irrelevance now.
